packages/runtime-core/src/componentEmits.ts

```diff
@@ -188,7 +188,9 @@ export function normalizeEmitsOptions(
   } else {
     extend(normalized, raw)
   }
-  return (comp.__emits = normalized)
+  // #2651 if normalized is empty and !raw should return null
+  return (comp.__emits =
+    !Object.keys(normalized).length && !raw ? null : normalized)
```

Comment:
I think it should rather change how the warning is performed

Reply:
I’m not even sure, I added a test case and it passed.

Reply:
Ideally this normalized version could be kept as an empty object, there is also the case of an empty `emits: {}`

Reply:
Thanks for your hint.I updated my solution.